src/kernels/helpers.ts

```diff
@@ -1867,7 +1867,7 @@ function handleExecuteSilentErrors(
 async function switchController(
     resource: Resource,
     serviceContainer: IServiceContainer
-): Promise<VSCodeNotebookController | undefined> {
+): Promise<{ controller: NotebookController; metadata: KernelConnectionMetadata } | undefined> {
```

Comment:
The object VSCodeNotebookController was too broad, we didn't need that whole object, all we need is the controller and the metadata.